elk/tests/test_extraction_output.py

```diff
@@ -1,17 +1,47 @@
 import pytest
 import pandas as pd
+from pathlib import Path
+from evaluate import evaluate
+import json
 
+default_config_path = Path(__file__).parent.parent / "default_config.json"
 
-class TestExtractionOutput:
-    def testExtractionOutput(self):
-        # Read the contents of the csv files into pandas dataframes
-        # TODO: Right now, the file paths are hardcoded. Improve on that.
-        correct_output_file = pd.read_csv(
-            "./test_data/deberta-v2-xxlarge-mnli_confusion_0.csv"
-        )
-        current_output_file = pd.read_csv(
-            "../evaluation_results/deberta-v2-xxlarge-mnli_confusion_0.csv"
-        )
+with open(default_config_path, "r") as f:
+    default_config = json.load(f)
+datasets = default_config["datasets"]
+models = default_config["models"]
+prefix = default_config["prefix"]
+models_layer_num = default_config["models_layer_num"]
 
-        # Compare the contents of the dataframes
-        assert correct_output_file.equals(current_output_file)
+
+def test_model_confusion_performance(model_name):
+
+    """
+    TODO: implement evaluation of the model with all datasets
+    # in progress
+    results_path = Path(__file__).parent.parent / f"{model_name}_confusion_0.csv"
+    seed = 0
+    # Evaluate the model
+    prompt = f"--model {model_name} --prefix normal  --dataset boolq
+     --num_data 1000 --save_dir {results_path} --seed {seed}"
+    evaluate(prompt)
+    """
+
+    # Read the contents of the csv files into pandas dataframes
+    benchmark_performance = pd.read_csv(
+        "../test_data/{}_confusion_0.csv".format(model_name)
+    )
+    current_performance = pd.read_csv(
+        "../evaluation_results/{}_confusion_0.csv".format(model_name)
+    )
+
+    # Compare the contents of the dataframes,
+    # make sure current performance are equal or better than the expected results
+    for index, row in benchmark_performance.iterrows():
+        assert row["accuracy"] <= current_performance.iloc[index]["accuracy"]
+        assert row["std"] <= current_performance.iloc[index]["std"]
```

Comment:
I guess we should maybe add some +/- epsilon (e.g having value 3)  to increase the tolerance a bit (?)

Reply:
Yeah probably. We should make sure the random seed is fixed and then just see empirically how big the discrepancy is with the current code